### PR TITLE
Parse Subprogram Instantiation Statement

### PR DESCRIPTION
**Pull request by Schottkyc137**
This PR only considers the syntactic parsing.
Analysis will come in a separate PR

### CODE REVIEW

File: vhdl_lang/src/analysis/overloaded.rs [hunk 1]
```diff
@@ -449,7 +449,7 @@ mod tests {
                 &fcall.pos,
                 &des,
                 &mut fcall.item.parameters,
-                SubprogramKind::Function(ttyp),
+                crate::analysis::overloaded::SubprogramKind::Function(ttyp),
```
Comment (kraigher) [inline vhdl_lang/src/analysis/overloaded.rs:452]: This change looks uncessary.

File: vhdl_lang/src/ast/display.rs [hunk 2]
```diff
@@ -863,6 +863,43 @@ impl Display for SubprogramDeclaration {
...
+}
+
+impl<T> Display for SeparatedList<T>
+where
+    T: Display,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        let mut first = true;
+        for assoc in &self.items {
+            if first {
```
Comment (kraigher) [inline vhdl_lang/src/ast/display.rs:892]: Suggestion: Factor out the common part of the `write!`

File: vhdl_lang/src/ast/search.rs [hunk 3]
```diff
@@ -1762,6 +1782,9 @@ impl std::fmt::Display for FoundDeclaration<'_> {
             FoundDeclaration::Function(ref value) => {
                 write!(f, "{value};")
             }
+            FoundDeclaration::SubprogramInstantiation(ref value) => {
+                write!(f, "{value};")
```
Comment (kraigher) [inline vhdl_lang/src/ast/search.rs:1786]: Question: Does this add ; twice? It is added in the Display trait as well.

**Review by kraigher — COMMENTED** [review]
Looks good. I just had a few minor comments.

File: vhdl_lang/src/analysis/overloaded.rs (hunk 1 above)
Comment (Schottkyc137) [reply vhdl_lang/src/analysis/overloaded.rs]: It conflicts with the `ast::SubprogramKind` wich is why I had to add this

**Review by kraigher — APPROVED** [review]

**Pull request merged**